unit-test/broker-protection-extract.js

```diff
@@ -0,0 +1,102 @@
+import {
+    createProfile
+} from '../src/features/broker-protection/actions/extract.js'
+
+describe('create profiles from extracted data', () => {
+    it('handles combined, single strings', () => {
+        const selectors = {
+            name: {
+                selector: '.name',
+                beforeText: ','
+            },
+            age: {
+                selector: '.name',
+                afterText: ','
+            }
+        }
+
+        const elementExamples = [
```

Comment:
So great to have test cases for this. It was such a guessing game when I first took the JSON files over!